extensions/tika/runtime/src/main/java/io/quarkus/tika/runtime/TikaConfiguration.java

```diff
@@ -12,11 +12,36 @@
 @ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
 public class TikaConfiguration {
     /**
-     * The path to the tika-config.xml
+     * Path to the tika-config.xml
      */
     @ConfigItem
     public Optional<String> tikaConfigPath;
 
+    /**
+     * Comma separated list of the parsers which must be supported.
+     * 
+     * Most of the document formats recognized by Apache Tika are supported by default but it affects
+     * the application memory and native executable sizes. One can list only the required parsers in
+     * tika-config.xml to minimize a number of parsers loaded into the memory, but using this
+     * property is recommended to achieve both optimizations.
+     * 
+     * Either the abbreviated or full parser class names can be used.
+     * At the moment only PDF parser can be listed using a reserved 'pdf' abbreviation.
+     * Using the custom abbreviations is recommended for all other parsers.
+     * 
+     * <pre>
+     * // Only PDF parser is required:
+     * tika-parsers = pdf
+     * // Only PDF and Java class parsers are required:
+     * tika-parsers = pdf,classparser
+     * classparser=org.apache.tika.parser.asm.ClassParser
```

Comment:
Are these examples correct?  It doesn't seem to match the properties in this class.

Reply:
@dmlloyd Thanks for spotting it the `tika-parsers` typo, should be just `parsers`.
The the `classparser` - this is a custom abbreviation and I'm going to restrict the `parsers` property containing only the abbreviations, either the ones recognized by the extension such as `pdf` or the custom one such as `classparser` (which may become promoted and become recognizable OOB) and not allow listing the class names in the `parsers` property directly:
the reason for it is, as I've mentioned above, I'd like to let users configure all the parsers (i.e the properties of the individual parsers) at the `application.properties` level without having to resort to typing `tika-config.xml`, 
So with the `parsers` property containing only the abbreviations (custom or built in ones), one would do:
```
quarkus.tika.parsers.pdf.sort-by-position=true
quarkus.tika.parsers.classparser.prop1=true
```

Reply:
Shouldn't this be a `Map` properly declared in the config object?

Reply:
@gsmet I have a different PR in the works to support this parser specific configuration, but as I've mentioned above, it requires [this issue](https://issues.apache.org/jira/browse/TIKA-2944) be resolved first, as under the hood the tika-config.xml will be generated but `TikaConfig` is unable at the moment to parse the properties without the XML type attributes and I'd like avoid users have to type that in themselves.

All this parser specific config is possible at the tika-config.xml level, [for example](https://github.com/apache/tika/blob/master/tika-parsers/src/test/resources/org/apache/tika/parser/pdf/tika-config.xml#L22). But this PR is a step toward letting users not to to use that XML config...

For the moment, for that another PR, I'm not adding a Map, and work with the `Config` provider directly to find out what is available for a given parser. If we add `Map` in `TikaConfiguration` then there will both the new `parsers` and this Map property will will be both optional/ignored if `tika-config-path` is available, so I have a minor preference to have only `parsers` here and doc how the parser-specific property can be set alongside the `parsers`.
In that PR I start with a `pdf.sort-by-position` - even though [TIKA-2944](https://issues.apache.org/jira/browse/TIKA-2944), effectively removing that tika-config.xml from the tests, because this property seems to be one of the most practical for the simple PDF invoices and I saw it being recommended few times. As such I reckon we can start with it with a minor property-specific check while keeping the code as generic as possible. I'll ask you and David to review that PR :-)